src/main/java/com/google/api/gax/core/FlowControlSettings.java

```diff
@@ -47,33 +48,35 @@ public static FlowControlSettings getDefaultInstance() {
   /** Maximum number of outstanding bytes to keep in memory before enforcing flow control. */
   @Nullable
   public abstract Integer getMaxOutstandingRequestBytes();
-  
+
   /** Is flow control enabled. Defaults to true. */
   public abstract boolean getIsEnabled();
-  
+
   /**
-   * Control whether the FlowController will fail with an exception or block the thread when flow
-   * control limits are exceeded.
+   * The behavior of FlowController when the specified limits are exceeded. Defaults to
+   * ThrowException.
    */
-  public abstract boolean getShouldFailOnFlowControlLimits();
+  public abstract LimitExceededBehavior getExceedLimitBehavior();
 
   public Builder toBuilder() {
     return new AutoValue_FlowControlSettings.Builder(this);
   }
 
   public static Builder newBuilder() {
-    return new AutoValue_FlowControlSettings.Builder().setIsEnabled(true);
+    return new AutoValue_FlowControlSettings.Builder()
+        .setIsEnabled(true)
+        .setExceedLimitBehavior(LimitExceededBehavior.ThrowException);
```

Comment:
I'd prefer for the setX and X enum to be the same - use `LimitExceededBehavior` for both.